pilot/pkg/xds/workloadentry.go

```diff
@@ -276,6 +270,10 @@ func workloadEntryFromGroup(name string, proxy *model.Proxy, groupCfg *config.Co
 	if proxy.Locality != nil {
 		entry.Locality = util.LocalityToString(proxy.Locality)
 	}
+	if proxy.Metadata.ProxyConfig != nil && proxy.Metadata.ProxyConfig.ReadinessProbe != nil {
+		annotations[status.WorkloadEntryHealthCheckAnnotation] = "true"
```

Comment:
Why set this annotation instead of using the condition?

Reply:
It is impossible to set status when you Create an object, you need to do `Create && UpdateStatus`. So there would be a race condition. I don't like this solution either though...

Reply:
I thought that if no health condition set, treat is as unhealthy.  Not sure if it makes sense. This requires us to change the istio-agent to send health info even there is no probe set.

Reply:
I think workload entry could even be for something that doesn't have a sidecar running potentially so I am not sure if we can always rely on status being set?

Reply:
You are right, i have not come up with a better way.